File: arrays/496. Next Greater Element I/nextGreaterElement.py
class Solution(object):
    def nextGreaterElement(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        ans = []

        for n in nums1:
            found = False

            for i in range(len(nums2)):
                if n == nums2[i]:
                    # Look for the next greater element amongst all the rest element to the right
                    for j in range(i + 1, len(nums2)):
                        if nums2[j] > n:
                            ans.append(nums2[j])
                            found = True
                            break

            # No next greater element found
            if not found:
                ans.append(-1)
        
        return ans

File: arrays/496. Next Greater Element I/nextGreaterElement.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var nextGreaterElement = function(nums1, nums2) {
    let ans = [];

    for (let n of nums1) {
        let found = false;

        for (let i = 0; i < nums2.length; i++) {
            if (n === nums2[i]) {
                for (let j = i + 1; j < nums2.length; j++) {
                    if (nums2[j] > n) {
                        ans.push(nums2[j]);
                        found = true;
                        break;
                    }
                }
            }
        }

        if (!found) {
            ans.push(-1);
        }
    }
    
    return ans;
};